messaging-client-databus/src/main/java/com/inmobi/databus/readers/LocalStreamCollectorReader.java

```diff
@@ -281,4 +281,60 @@ protected Date getTimeStampFromCollectorStreamFile(FileStatus file) {
     }
     return null;
   }
+
+  protected Long doRecursiveSizing(Path dir, PathFilter pathFilter) throws IOException {
+    Long pendingSize =0l;
+    FileStatus[] fileStatuses = fsListFileStatus(dir, pathFilter);
+    if (fileStatuses == null || fileStatuses.length == 0) {
+      LOG.debug("No files in directory:" + dir);
+    } else {
+      for (FileStatus file : fileStatuses) {
+        if (file.isDir()) {
+          pendingSize += doRecursiveSizing(file.getPath(), pathFilter);
+        } else {
+          try {
+            Date currentTimeStamp = LocalStreamCollectorReader.
+                    getDateFromStreamFile(streamName, file.getPath().getName());
+            if (stopTime != null && stopTime.before(currentTimeStamp)) {
+              continue;
+            } else {
+              pendingSize += file.getLen();
+            }
+          } catch (Exception e) {
+            LOG.error("Exception while getting time from File "+file.getPath().toString(),e);
+            throw new IOException(e);
+          }
+        }
+      }
+    }
+    return pendingSize;
+  }
+
+  public Long getPendingSize() throws IOException {
+    Long pendingSize = 0L;
+    Calendar current = Calendar.getInstance();
+    Date now = current.getTime();
+    current.setTime(buildTimestamp);
+    // stop the file listing if stop date is beyond current time
+    while (current.getTime().before(now)) {
+      Path dir = getMinuteDirPath(streamDir, current.getTime());
+      // Move the current minute to next minute
+      current.add(Calendar.MINUTE, 1);
```

Comment:
I wanted to omit the current file from the remaining size. may be the current directory is almost read already, why to avoid that to the pending size.